docs/bundler.md

```diff
@@ -0,0 +1,141 @@
+# bundler
+
+<https://bundler.io/>
+
+## Prerequisites
+
+To use Cachi2 with Bundler locally, ensure you have Ruby and Bundler installed
+on your system.
+
+```bash
+sudo dnf install rubygem-bundler
+```
+
+Or use the official Ruby image for example from Docker Hub:
+
+```bash
+podman run --rm -it docker.io/library/ruby:latest bash
+```
+
+Then ensure you have both, **Gemfile** and **Gemfile.lock** in your project
+directory. Anything outside of the **Gemfile.lock** will be ignored. We parse
+the **Gemfile.lock** to pre-fetch all dependencies specified in that file.
+Please note, that we don't use any Bundler commands to pre-fetch dependencies.
+Instead, we use native mechanisms from the Ruby standard library to parse
+the **Gemfile.lock** and download the dependencies to the specified output
+directory.
+
+## Usage
+
+Run the following command in your terminal to pre-fetch your project's
+dependencies. The command will download all dependencies specified in the
+**Gemfile.lock** to the specified output directory.
+
+```bash
+cd path-to-your-ruby-project
+cachi2 fetch-deps --source ./path-to-your-ruby-project --output ./cachi2-output bundler
+```
+
+In addition, it will prepare the necessary environment variables and
+configuration files for the build phase. See the following section for more
+information.
+
+### Configuration
+
+Bundler uses an unorthodox system when dealing with configuration options.
+The highest precedence is given to the config file, and then to the environment
+variables. See the official documentation:
+<https://bundler.io/v2.5/man/bundle-config.1.html#DESCRIPTION>>
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment
```

Comment:
This paragraph should be placed in a subsection (or maybe a dedicated section) called e.g. "Known issues" or "Known limitations" and placed at the very end of the configuration paragraph (or at the end of the document below _Unsupported features_).